aptos-move/mvhashmap/src/versioned_group_data.rs

```diff
@@ -219,6 +265,17 @@ impl<T: Hash + Clone + Debug + Eq + Serialize, V: TransactionWrite> VersionedGro
                 (Vacant(entry), Creation) => {
                     entry.insert(v.clone());
                 },
+                (Occupied(mut entry), Creation) => {
+                    if entry.get().write_op_kind() == Deletion {
```

Comment:
We may be able to use the match guard here: https://doc.rust-lang.org/beta/reference/expressions/match-expr.html#match-guards, i.e. fold the if into the match arm itself. if possible that's nice because the else branch would also go away.

Reply:
How can committed group entry be Deletion? whenever we commit Deletion we delete the entry? (see just above, line 260). @igor-aptos what was the case when this happened?

Reply:
Done.

Reply:
actually, you are right, this cannot be the case any more, we can remove the
                (Occupied(mut entry), Creation) if entry.get().write_op_kind() == Deletion => {
match case

previously, and for regular resources - if value doesn't exist - Deletion is stored as base value.
now we have "TagNotFound" and don't need to store deletion to know not to fetch again from storage, so this path is never reached.